### PR TITLE
Simple Todo List

### PR DESCRIPTION
**Pull request by ghulamhafeez**
## Simple Todo List

- Added search functionality in todo list

### Search Bar Gif
todoSearch

- Created unit test & also clear testing our component
### Unit Testing Screenshot
image

### CODE REVIEW

File: src/App.test.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
-import { render, screen } from '@testing-library/react';
-import App from './App';
+// import { render, screen } from '@testing-library/react';
```
Comment (chusama-dev) [inline src/App.test.js:1]: Please remove the comments

File: src/services/Todoservices.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// import {BaseUrl} from './constant'
+
+
+export const getTodos =()=>{
```
Comment (Monfernape) [inline src/services/Todoservices.js:4]: Nice

File: src/views/Todo-Components/TodoFilter.test.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { render ,screen} from "@testing-library/react";
+import TodoFilter from "./TodoFilter";
+
+test ("Search input  testing",()=>{
+    render(<TodoFilter searchValue={"abc"}/>)
+    const searchInput = screen.getByTestId("search-input");
+    expect(searchInput).toBeInTheDocument();
+    expect(searchInput).toHaveValue("abc")
```
Comment (Monfernape) [inline src/views/Todo-Components/TodoFilter.test.js:8]: Super cool

File: src/views/Todo-Components/TodoForm.js [hunk 4]
```diff
@@ -1,21 +1,52 @@
 import React from "react";
+import { useFormik } from "formik";
+import * as Yup from "yup";
+
+export const TodoForm = ({ todo, onUpdate, onAdd, isEditMode }) => {
+  const schema = Yup.object().shape({
+    todo: Yup.string().required("Fill Todo Field"),
```
Comment (Monfernape) [inline src/views/Todo-Components/TodoForm.js:7]: Let's try to name error messages like this: `Name is required`

File: src/views/Todo-Components/TodoForm.js [hunk 5]
```diff
@@ -1,21 +1,52 @@
...
+  });
+
+  const { handleBlur, handleChange, values, handleSubmit, errors } = useFormik({
+    initialValues: {
+      todo: todo,
+    },
+
+    validationSchema: schema,
+    onSubmit: (values, { resetForm }) => {
+      isEditMode ? onUpdate(values.todo) : onAdd(values.todo);
```
Comment (Monfernape) [inline src/views/Todo-Components/TodoForm.js:17]: Nice stuff

File: src/views/Todo-Components/TodoForm.js [hunk 6]
```diff
@@ -1,21 +1,52 @@
...
+      resetForm({ todo: "" });
+    },
+    enableReinitialize: true,
```
Comment (Monfernape) [inline src/views/Todo-Components/TodoForm.js:20]: Remember this, it's key to refresh formik state.

File: src/views/Todo-Components/TodoForm.test.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    const UpDate = screen.getByText(/UpDate/i)
+    expect(UpDate).toBeInTheDocument();
+})
+
+test ("Add button working Test" , ()=>{
+    render(<TodoForm onAdd={onAdd} />)
+    const AddTest = screen.getByTestId("AddTest")
+    expect(AddTest).toBeInTheDocument();
+    fireEvent.click(AddTest);
+  expect(onAdd).toHaveBeenCalled();
```
Comment (Monfernape) [inline src/views/Todo-Components/TodoForm.test.js:31]: Amazing

File: src/views/Todo-Components/TodoList.test.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+test("Delete Button  working Test", () => {
+  render(
+    <TodoList
+      onEdit={onEdit}
+      onDelete={onDelete}
+      onToggleCompletion={onToggleCompletion}
+      todos={todos}
+    />
+  );
+  const deleteButton = screen.getByTestId(`delete-button-${todos[0].id}`);
```
Comment (Monfernape) [inline src/views/Todo-Components/TodoList.test.js:34]: Do you understand this functionality?

**Review by Monfernape — APPROVED** [review]
LGTM

File: src/views/Todo-Components/TodoFilter.test.js (hunk 3 above)
Comment (ghulamhafeez) [reply src/views/Todo-Components/TodoFilter.test.js:8]: Thanks

File: src/views/Todo-Components/TodoForm.js (hunk 4 above)
Comment (ghulamhafeez) [reply src/views/Todo-Components/TodoForm.js:7]: ok

File: src/views/Todo-Components/TodoForm.js (hunk 5 above)
Comment (ghulamhafeez) [reply src/views/Todo-Components/TodoForm.js:17]: Thanks

File: src/views/Todo-Components/TodoForm.js (hunk 6 above)
Comment (ghulamhafeez) [reply src/views/Todo-Components/TodoForm.js:20]: ok

**Pull request merged**